src/components/SeedInfo/SeedInfoViews/HolyMountain.tsx

```diff
@@ -135,16 +136,36 @@ const PacifistChest: FC<IPacifistChestProps> = ({ items }) => {
 // TODO: Extract this into it's own file to decouple
 const Shop = ({ type, handleOpenShopInfo, favoriteSpells }) => {
   const Icon = type === IShopType.wand ? WandIcon : LightBulletIcon;
+
+  // group duplicate favorite spells and count occurrences
+  const favoriteSpellsGrouped : Map<string, number> = favoriteSpells.reduce((acc, spell) => {
+    acc.set(spell, (acc.get(spell) ?? 0) + 1); return acc;
+  }, new Map<string, number>());
+  
+
+  // add favorite spell icons with badges for counts
+  const favSpellIcons = Array.from(favoriteSpellsGrouped.entries()).map(([spell, count]) => (
```

Comment:
When there are lots of spells here - worst case 10 or even more - it'll be very ugly. There are two (and a bonus third) ways to work with this. 
1. We can limit the number of shown favorite spells. From the screenshot, 3 looks like a good number. Will cover 90% of use-cases. Simple, easy, but if someone has many favorites, that wouldn't work.
2. Show 3, but change the 3rd into a ... or + or somehting if there are 4 or more. That way we can indicate that there are more that we are not showing
3. Just ignore it and ship